lib/mongo/scrollable.rb

```diff
@@ -16,21 +16,37 @@ def scroll(cursor_or_type = nil, options = nil, &_block)
       cursor_options = { field_name: scroll_field, direction: scroll_direction }.merge(options)
       cursor = cursor && cursor.is_a?(cursor_type) ? cursor : cursor_type.new(cursor, cursor_options)
       raise_mismatched_sort_fields_error!(cursor, cursor_options) if different_sort_fields?(cursor, cursor_options)
-      # make a view
-      view = Mongo::Collection::View.new(
-        view.collection,
-        view.selector.merge(cursor.criteria),
-        sort: (view.sort || {}).merge(_id: scroll_direction),
-        skip: skip,
-        limit: limit
-      )
+
+      records = nil
+      if cursor.type == :previous && limit
```

Comment:
The `&& limit` is suspect. I think it's incorrect if `cursor.type == :previous` but `limit = nil`.

IMO `cursor.type` should always be assigned to either `:previous` or `:next`. Throw an exception inside deserialization when its neither of those values.

Here:

```
case cursor.type:
when :next, nil: ...
when :previous ...
else
   raise an error here
```

Reply:
I removed the `&& limit`. It wasn't needed anymore.

I still have to raise an error when there is no `type` options.

Reply:
I added an error when the type is other than :next of :previous.